engine/graphics/src/vulkan/graphics_vulkan.cpp

```diff
@@ -1628,6 +1628,11 @@ namespace dmGraphics
             vk_scissor.offset.y = 0;
 
             VkResult res = CreateGraphicsPipeline(vk_device, vk_scissor, vk_sample_count, pipelineState, program, vertexDeclaration, vertexDeclarationCount, rt, &new_pipeline);
+            if (res == VK_ERROR_INITIALIZATION_FAILED)
+            {
+                dmLogError("Failed to create VkPipeline");
+                return 0;
```

Comment:
We do wish to catch this earlier (build time) as the error isn't really helpful (See https://github.com/defold/defold/issues/10410)
But the main thing for now is to prevent an unnecessary assert.